sdk/eventhub/Microsoft.Azure.EventHubs.Processor/src/AzureStorageCheckpointLeaseManager.cs

```diff
@@ -79,6 +81,15 @@ internal void Initialize(EventProcessorHost host)
                 MaximumExecutionTime = TimeSpan.FromSeconds(this.LeaseRenewInterval.TotalSeconds)
             };
 
+            // Storage suggests to use at least 10 seconds timeout at list-blobs calls.
+            this.listBlobsRequestOptions = new BlobRequestOptions()
+            {
+                ServerTimeout = this.defaultRequestOptions.ServerTimeout,
+                MaximumExecutionTime = this.defaultRequestOptions.MaximumExecutionTime
+            };
+            this.listBlobsRequestOptions.ServerTimeout = this.listBlobsRequestOptions.ServerTimeout < MinListBlobsTimeout ?
```

Comment:
It may be helpful to call out that the `MaximumExecutionTime` is guaranteed to be at least 16 seconds due to enforcement in the `PartitionManagerOptions`;  I had to jump around a bit before that was clear to me.

Reply:
Actually, that is lease duration with 15 seconds minimum value which is also enforced by Storage. `MaximumExecutionTime` is set from `PartitionManagerOptions.LeaseRenewInterval` where a developer is free to pick a very small number and this has been the issue at list-blobs API call in one of the incidents.